File: javascript/difference-of-squares/difference-of-squares.js
//@ts-check

export class Squares {
  /**
   * @param {number} n
   */
  constructor(n) {
    /** @type {number} */
    this.n = n;
  }

  get sumOfSquares() {
    return (this.n * (this.n + 1) * (2 * this.n + 1)) / 6;
  }

  get squareOfSum() {
    return ((this.n * (this.n + 1)) / 2) * ((this.n * (this.n + 1)) / 2);
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SumOfSquares finds the sum of the squares of the first N natural numbers.
func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

// SquareOfSum finds the square of the sum of the first N natural numbers.
func SquareOfSum(n int) int {
	return (n * (n + 1) / 2) * (n * (n + 1) / 2)
}

// Difference finds the difference between SquareofSum and SumOfSquares.
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
